src/dso/dso_transport.hpp

```diff
@@ -0,0 +1,240 @@
+ *       derived from this software without specific prior written permission.
+ *
+ *    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *    AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *    IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *    ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *    LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *    CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *    SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *    INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *    CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *    ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *    POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ *   This file includes definitions for DSO agent.
+ */
+
+#ifndef OTBR_AGENT_DSO_TRANSPORT_HPP_
+#define OTBR_AGENT_DSO_TRANSPORT_HPP_
+
+#if OTBR_ENABLE_DNS_DSO
+
+#include <arpa/inet.h>
+#include <map>
+#include <netinet/in.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <vector>
+
+#include <cassert>
+#include <list>
+
+#include "common/code_utils.hpp"
+#include "mbedtls/ctr_drbg.h"
+#include "mbedtls/debug.h"
+#include "mbedtls/entropy.h"
+#include "mbedtls/error.h"
+#include "mbedtls/net_sockets.h"
+#include "mbedtls/ssl.h"
+#include "openthread/logging.h"
+#include "openthread/message.h"
+#include "openthread/platform/dso_transport.h"
+
+namespace otbr {
+namespace Dso {
+
+// TODO: Support DNS-over-TLS
+/**
+ * @addtogroup border-router-dso
+ *
+ * @brief
+ *   This module includes definitions for DSO agent.
+ *
+ * @{
+ */
+
+class DsoAgent
+{
+private:
+    class DsoConnection;
+
+public:
+    /**
+     * This constructor initializes the DsoAgent instance.
+     *
+     */
+    explicit DsoAgent(void);
+
+    /**
+     * This method enables listening for DsoAgent.
+     *
+     * @param[in] aInstance  The OT instance.
+     *
+     */
+    void Enable(otInstance *aInstance);
+
+    /**
+     * This method disables listening for DsoAgent.
+     *
+     * @param[in] aInstance  The OT instance.
+     *
+     */
+    void Disable(otInstance *aInstance);
+
+    /**
+     * This method enables/disables listening for DsoAgent.
+     *
+     * @param[in] aInstance  The OT instance.
+     * @param[in] aEnabled   A boolean indicating whether to enable listening.
+     *
+     */
+    void SetEnabled(otInstance *aInstance, bool aEnabled);
+
+    /**
+     * This method finds the DsoConnection corresponding to the given otPlatDsoConnection.
+     *
+     * @param[in] aConnection  A pointer to the otPlatDsoConnection.
+     *
+     * @returns  A pointer to the matching DsoConnection object.
+     *
+     */
+    DsoConnection *Find(otPlatDsoConnection *aConnection);
```

Comment:
nit:
```suggestion
    DsoConnection *FindConnection(otPlatDsoConnection *aConnection);
```

Same for other Connection specific methods.